### PR TITLE
Tlportal 79: log throttling messages from data provider

### PR DESCRIPTION
**Pull request by dlhaines**
Log any throttling messages from the data provider and have check url respond correctly to throttling messages.

### CODE REVIEW

File: server/local/studentdashboard-LOCALHOST.yml [hunk 1]
```diff
@@ -7,6 +7,8 @@
 application_name: SD-QA
 # use file based data provider if not commented.
 data_provider_file_directory: server/test-files
+# used for testing response to throttled check url requests.
+#data_provider_file_uniqname: test-throttling-terms
```
Comment (zqian) [inline server/local/studentdashboard-LOCALHOST.yml]: Do those newly added params need to be uncommented out?

File: server/courselist.rb [hunk 2]
```diff
@@ -679,6 +681,13 @@ def self.getURLUniqname(url)
 
   #################### Data provider functions #################
 
+  # If a response response from the provider is throttled then log that fact.
+  SERVICE_UNAVAILABLE = "503"
+
+  def logIfUnavailable(response,msg)
+    logger.warn("provider response throttled or unavailable for: #{msg}") if SERVICE_UNAVAILABLE.casecmp(response.meta_status.to_s).zero?
```
Comment (zqian) [inline server/courselist.rb]: Not sure whether we need to use casecmp if we are just comparing status code.

Comment (dlhaines) [inline server/courselist.rb]: We don't need the casecmp but IMHO it's more efficient for developers to always put it in for string compare.  If nothing else this avoids having case bugs if the string, at some point, includes characters.  There may be circumstances when case is important for a compare but respecting case is more likely to be a problem than ignoring case.

File: server/local/studentdashboard-LOCALHOST.yml (hunk 1 above)
Comment (dlhaines) [inline server/local/studentdashboard-LOCALHOST.yml]: That comment is meant to preserve information to make testing easier down the road.

**Comment by lsloan** [comment]
I like it.  :shipit:

File: server/courselist.rb (hunk 2 above)
Comment (zqian) [inline server/courselist.rb]: I agree!

**Pull request merged**